com.unity.render-pipelines.core/ShaderLibrary/EntityLighting.hlsl

```diff
@@ -286,6 +286,24 @@ real3 SampleSingleLightmap(TEXTURE2D_PARAM(lightmapTex, lightmapSampler), float2
     return illuminance;
 }
 
+real3 SampleSingleLightmap(TEXTURE2D_ARRAY_PARAM(lightmapTex, lightmapSampler), float2 uv, float slice, float4 transform, bool encodedLightmap, real4 decodeInstructions)
```

Comment:
The strategy I usually use when we need to include multiple times the same function but with different sampler, is to put all functions in a separate file and include it multiple time with predefine macro

i.E

#define SAMPLER_LIGHTMAP SAMPLE_TEXTURE2D_ARRAY
#include "EntityLightingTemplate.hlsl"

#undef SAMPLER_LIGHTMAP 
#define SAMPLER_LIGHTMAP  SAMPLE_TEXTURE2D
#include "EntityLightingTemplate.hlsl"

Avoid code duplication as here  I am afraid in case of change of lightmap encoding we forget to update all the function.

But I let you judge if you want to do this change.

Reply:
Good strategy, I will do this.